assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -154,6 +166,58 @@ const baseActions = {
 
 		return { response, error };
 	},
+
+	/**
+	 * @param {string}          slug                         Module slug.
+	 * @param {Object}          settings                     Module settings.
+	 * @param {string}          [settings.name]              Optional. Module name. Default is the slug.
+	 * @param {string}          [settings.description]       Optional. Module description. Default empty string.
+	 * @param {string}          [settings.icon]              Optional. Module icon. Default empty string.
+	 * @param {number}          [settings.order]             Optional. Numeric indicator for module order. Default 10.
+	 * @param {string}          [settings.homepage]          Optional. Module homepage URL. Default empty string.
+	 * @param {boolean}         [settings.internal]          Optional. Whether the module is considered internal. Default false.
```

Comment:
Should this be a public/documented param? Would we ever want modules to use this setting? Should we at least say it's not for 3rd-parties?